hack/table-gen/table-gen.go

```diff
@@ -0,0 +1,264 @@
+	kind := getElement(obj, "spec", "names", "kind")
+	group := getElement(obj, "spec", "group")
+	CRDKind = kind.(string)
+	CRDGroup = group.(string)
+
+	for _, version := range versions.([]interface{}) {
+		name := getElement(version, "name")
+		APIVersion = name.(string)
+
+		spec := getElement(version, "schema", "openAPIV3Schema", "properties", "spec")
+		mergeMaps(docElements, generateElementDoc(elementsToSkip, spec, "spec", ""))
+
+		status := getElement(version, "schema", "openAPIV3Schema", "properties", "status")
+		mergeMaps(docElements, generateElementDoc(elementsToSkip, status, "status", ""))
+	}
+
+	var doc []string
+	for _, propName := range sortKeys(docElements) {
+		doc = append(doc, docElements[propName])
+	}
+
+	doc = append([]string{
+		"<!-- " + CRDKind + " " + APIVersion + " " + CRDGroup + " -->",
+		"| Parameter         | Description                                   |",
+		"| ---------------------------------------- | ---------|",
+	}, doc...)
+
+	return strings.Join(doc, "\n")
+}
+
+// generates table row out of some CRD element. Returns resulting map with path of an element and table row
+func generateElementDoc(elementsToSkip map[string]bool, obj interface{}, name string, parentPath string) map[string]string {
+	result := map[string]string{}
+	element := obj.(map[string]interface{})
+	elementType := element["type"].(string)
+	description := ""
+	if d := element["description"]; d != nil {
+		description = d.(string)
+	}
+
+	fullName := fmt.Sprintf("%s%s", parentPath, name)
+	skipWithAncestors, shouldBeSkipped := elementsToSkip[fullName]
+	if shouldBeSkipped && skipWithAncestors {
+		return result
+	}
+
+	if !shouldBeSkipped {
+		result[fullName] = generateTableRow(fullName, description, name)
+	}
+
+	if elementType == "object" {
+		mergeMaps(result, generateObjectDoc(elementsToSkip, element, name, parentPath))
+	}
+
+	if elementType == "array" {
+		mergeMaps(result, generateArrayDoc(elementsToSkip, element, name, parentPath))
+	}
+
+	return result
+}
+
+// generates table row out of CRD object with type object. Returns resulting map with path of an element and table row
+func generateObjectDoc(elementsToSkip map[string]bool, element map[string]interface{}, name string, parentPath string) map[string]string {
+	result := map[string]string{}
+	properties := getElement(element, "properties")
+	if properties == nil {
+		return result
+	}
+
+	propMap := properties.(map[string]interface{})
+	for _, propName := range sortKeys(propMap) {
+		mergeMaps(result, generateElementDoc(elementsToSkip, propMap[propName], propName, parentPath+name+"."))
+	}
+	return result
+}
+
+// generates table row out of CRD object with type array. Returns resulting map with path of an element and table row
+func generateArrayDoc(elementsToSkip map[string]bool, element map[string]interface{}, name string, parentPath string) map[string]string {
+	result := map[string]string{}
+	properties := getElement(element, "items")
```

Comment:
I think it is more suitable to name this variable as `items` instead of `properties`. I guess the `properties` name was just copied from the function above `properties := getElement(element, "properties")`